File: CivitasRuby/Sorpresa_IrCasilla.rb
module Civitas
    class Sorpresa_IrCasilla < Sorpresa
        public_class_method :new
        def initialize(tablero,valor,texto)

            super(texto)
            @tablero=tablero
            @valor=valor
            
        end

        def aplicarAJugador(actual, todos)

            if jugadorCorrecto(actual,todos)
                informe(actual,todos)
                casillaActual=todos[actual].numCasillaActual
                tirada=@tablero.calcularTirada(casillaActual,@valor)
                todos[actual].moverACasilla(@tablero.nuevaPosicion(casillaActual,tirada))
                @tablero.getCasilla(@valor).recibeJugador(actual, todos)
            end

        end

        def toString
            return @texto
        end
    end
end

File: CivitasJava/civitas/Sorpresa_IrCasilla.java
package civitas;

import java.util.ArrayList;

public class Sorpresa_IrCasilla extends Sorpresa
{
    private int valor;
    private Tablero tablero;

    Sorpresa_IrCasilla(Tablero tablero,int valor, String texto)
    {
        super(texto);
        this.tablero=tablero;
        this.valor=valor;
    }

    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos)
    {
        if(super.jugadorCorrecto(actual, todos))
        {
            super.informe(actual,todos);

            int casillaActual=todos.get(actual).getNumCasillaActual();
            int tirada=tablero.calcularTirada(casillaActual, valor);
            todos.get(actual).moverACasilla(tablero.nuevaPosicion(casillaActual, tirada));
            tablero.getCasilla(valor).recibeJugador(actual, todos);
        }
    }

    @Override
    public String toString()
    {
        return super.texto;
    }
}